packages/usa-header/src/styles/_usa-megamenu.scss

```diff
@@ -9,18 +9,28 @@
   display: block;
   position: absolute;
 
-  // retrieve the unit value of $theme-header-max-width, depending on type of value added
-  $mw: smart-quote($header-max-width);
-  @if meta.type-of($header-max-width) == "string" {
-    $mw: units($header-max-width);
-  }
+  // Bypass width calculation if $header-max-width is set to none
+  @if $header-max-width == "none" {
+    @if $type == "standard" {
+      // standard needs the $theme-site-margins-width to accommodate different paddings/structure on #basic-mega-nav-section-two
+      width: units($theme-site-margins-width);
+    } @else if $type == "extended" {
+      width: 0;
+    }
+  } @else {
+    // retrieve the unit value of $theme-header-max-width, depending on type of value added
+    $mw: smart-quote($header-max-width);
+    @if meta.type-of($header-max-width) == "string" {
+      $mw: units($header-max-width);
+    }
 
-  // subtract half the value of the viewport width from half the value of the submenu width
-  // standard needs the additional $theme-site-margins-width to accommodate different paddings/structure on #basic-mega-nav-section-two
-  @if $type == "standard" {
-    width: calc(50vw - $mw / 2 + units($theme-site-margins-width));
-  } @else if $type == "extended" {
-    width: calc(50vw - $mw / 2);
+    // subtract half the value of the viewport width from half the value of the submenu width
+    // standard needs the additional $theme-site-margins-width to accommodate different paddings/structure on #basic-mega-nav-section-two
+    @if $type == "standard" {
+      width: calc(50vw - $mw / 2 + units($theme-site-margins-width));
+    } @else if $type == "extended" {
+      width: calc(50vw - $mw / 2);
+    }
   }
 }
```

Comment:
Maybe we could simplify this and only check for `"standard"` once by tweaking the original logic? Since "none" is a valid token, we just need to make sure the value isn't set to none so the math works properly.

```scss
  // Retrieve the unit value of $theme-header-max-width, depending on type of value added
  $mw: smart-quote($header-max-width);
  @if meta.type-of($header-max-width) == "string" {
    $mw: units($header-max-width);
  }

  // Subtract half the value of the viewport width from half the value of the submenu width
  //
  // Standard needs the additional $theme-site-margins-width to accommodate
  // different paddings/structure on #basic-mega-nav-section-two
  @if $theme-header-max-width != "none" {
    @if $type == "standard" {
      width: calc(50vw - $mw / 2 + units($theme-site-margins-width));
    } @else if $type == "extended" {
      width: calc(50vw - $mw / 2);
    }
  }
```

Reply:
@mejiaj I like this solution - much cleaner! The only issue is `outer-megamenu` needs to return unique values when `$theme-header-max-width` is set to `none`. I have cleaned up the original solution to (hopefully) be more readable. Let me know if you see a better solution.

Reply:
@amyleadem I might be misunderstanding. I wasn't able to see a difference in visual display or code from [c3d08f8](https://github.com/uswds/uswds/pull/5072/commits/c3d08f850b135753621829af567190f84700e976) and the code example above.

If max-width is set to `none` then there should be no need for pseudo elements before/after the dropdown menu. Maybe we can quickly sync next week.

Reply:
@mejiaj I might be misunderstanding as well, so happy to chat next week. For reference, here is what I am trying to account for in with [line 29](https://github.com/uswds/uswds/pull/5072/files#diff-c820f85b4a911dbfcc6b90a3ffa0d73c429033a5d52cc034ba6cf82a057993b6R29). 

In the standard megamenu, `usa-megamenu`  is inside `usa-nav-container`, which has a padding-x value of `$theme-site-margins-width`. Because of this structure and padding, `usa-megamenu` does not extend to the edge of the window, even when the `max-width` is set to `none`. To account for this, it needs a pseudo element at least the width of `$theme-site-margins-width`. This is why I added those extra conditionals. 

<img width="1551" alt="image" src="https://user-images.githubusercontent.com/93996430/215210415-e4b59d8e-5f81-45da-be57-0ededd68b473.png">

Just wanted to capture that before the weekend. Let's chat more on Monday!

PS - I set the storybook layout to `fullscreen` in these examples so that it is clear where the megamenu ends.